libminifi/include/core/logging/LoggerBase.h

```diff
@@ -109,6 +140,7 @@ class LoggerBase : public Logger {
 
  private:
   std::atomic<int> max_log_size_{LOG_BUFFER_SIZE};
+  std::vector<std::function<void(LOG_LEVEL level, const std::string&)>> log_callbacks_;
```

Comment:
do we need multiple callbacks? If not, I'd say keep it simple and only have one.